### PR TITLE
Found the leela reward bug

### PR DESCRIPTION
**Pull request by sej2020**
Simple bug, very difficult to locate. I'm pretty sure the reward representation is perfect now for the leela encodings.

### CODE REVIEW

File: tests/unit/utils/test_leela_encodings.py [hunk 1]
```diff
@@ -9,40 +9,71 @@
...
+            13. Bd4 d5 14. Ba4 h4 15. Qf3 h3 16. g3 Bxd4 17. exd4 O-O 18. Qf4 f6
+            19. a3 b3 20. cxb3 Qb6 21. b4 Nb7 22. g4 Bd7 23. Nc3 Rad8 24. Rae1 a6
+            25. Re7 Rf7 26. Rfe1 Kf8 27. Rxf7+ Kxf7 28. g5 a5 29. g6+ Kxg6 30. Re7 axb4
+            31. Qg3+ Kf5 32. Bd1 Qxd4 33. Qxh3+ Kg5 34. Qh5+ Kf4 35. Ne2#"""
+
+
+@pytest.fixture(scope="module")
+def stockfish_metric1():
+    return StockfishMetric()
+
```
Comment (Xmaster6y) [inline tests/unit/utils/test_leela_encodings.py:36]: Should be a generator using yield

File: tests/unit/utils/test_leela_encodings.py [hunk 2]
```diff
@@ -9,40 +9,71 @@
...
+
+@pytest.fixture(scope="module")
+def stockfish_metric2():
+    """
+    A second stockfish metric is needed because of difficulties deleting the first
+        metric in the testing environment
+    """
     return StockfishMetric()
```
Comment (Xmaster6y) [inline tests/unit/utils/test_leela_encodings.py:45]: Generator

File: src/utils/leela_encodings.py [hunk 3]
```diff
@@ -152,10 +152,10 @@ def encode_seq(
     us, them = chess.WHITE, chess.BLACK
     for alg_move in seq.split():
         board_tensors[us].append(board_to_tensor(board, (us, them)))
-        if position_evaluator is not None:
-            position_evaluations[us].append(position_evaluator(board, (us, them)))
         if alg_move.endswith("."):
             continue
```
Comment (Xmaster6y) [inline src/utils/leela_encodings.py:156]: This should be the first test in the loop, right?

Comment (sej2020) [reply src/utils/leela_encodings.py:156]: oh you're right, I was so focused on the position evaluations, I didn't notice that this should go above the board_tensors[us].append() line as well. I'll make a commit for this

**Pull request merged**